### DISCUSSION TITLE
Argument missing for parameter "self" when plot with `xarray`

### DISCUSSION THREAD

**Discussion by singledoggy** [microsoft/pyright#8203] [Q&A] [answered]
I have been using xarray for several years and recently began using Pyright. However, I encountered the following error: `Argument missing for parameter 'self'` Although my code still executes normally, I am curious about the cause of this error. Is this behavior expected?
 I asked this question there but I think according to the comments in the function `class UncachedAccessor(Generic[_Accessor]):`, they've tried to aviod this error.

I've noticed the disscussion [here](https://github.com/microsoft/pyright/discussions/6669_) but I can't figure out the reason.

```
import xarray as xr
import matplotlib.pyplot as plt
data = xr.DataArray([1, 2, 3], dims=['x'])
fig, ax= plt.subplots(ncols=1, nrows=1)
data.plot(ax=ax)
```
It seems to related to the codes here:
```
    plot = utils.UncachedAccessor(DataArrayPlotAccessor)

```
```
class UncachedAccessor(Generic[_Accessor]):
    """Acts like a property, but on both classes and class instances

    This class is necessary because some tools (e.g. pydoc and sphinx)
    inspect classes for which property returns itself and not the
    accessor.
    """

    def __init__(self, accessor: type[_Accessor]) -> None:
        self._accessor = accessor

    @overload
    def __get__(self, obj: None, cls) -> type[_Accessor]: ...

    @overload
    def __get__(self, obj: object, cls) -> _Accessor: ...

    def __get__(self, obj: None | object, cls) -> type[_Accessor] | _Accessor:
        if obj is None:
            return self._accessor

        return self._accessor(obj)  # type: ignore  # assume it is a valid accessor!
```
Here is my `xarray` version:
```
xarray                    2023.2.0           pyhd8ed1ab_0    conda-forge
```

_Originally posted by @singledoggy in https://github.com/pydata/xarray/discussions/9150_

**Comment by erictraut** [comment, marked as answer]
Pyright is behaving correctly here. Not surprisingly, mypy generates the same error.

The problem appears to be in the `xarray` library here:

```python
class DataArrayPlotAccessor:
    ...
    @functools.wraps(dataarray_plot.plot, assigned=("__doc__", "__annotations__"))
    def __call__(self, **kwargs) -> Any:
        return dataarray_plot.plot(self._da, **kwargs)
```

**Reply by singledoggy** [reply]
Thank you for your prompt reply. I attempted to add some comments using `# type: ignore` but did not succeed in addressing the issue. Do you have additional suggestions for resolving this? Considering they've tried this workaround multiple times in their library, I assume they are not experts in using type checkers.

**Comment by singledoggy** [comment]
> Looks like [python/mypy#17166](https://github.com/python/mypy/issues/17166)
